map/src/map/layers/LocalClientTrackLayer.js

```diff
@@ -0,0 +1,58 @@
+import React, {useContext, useEffect, useState} from "react";
+import AppContext from "../../context/AppContext";
+import {useMap} from "react-leaflet";
+import L from "leaflet";
+import MarkerIcon from "../MarkerIcon";
+
+
+export default function LocalClientTrackLayer() {
+    const ctx = useContext(AppContext);
+    const map = useMap();
+
+    const [layers, setLayers] = useState([]);
+
+    const markerOptions = {
+        startIcon: MarkerIcon({bg: 'blue'}),
+        endIcon: MarkerIcon({bg: 'red'}),
+        wptIcons: {
+            '': MarkerIcon({bg: 'yellow'}),
+        }
+    };
+
+    function removeLayerFromMap(layer) {
+        map.removeLayer(layer.layer);
+        layers.splice(layers.indexOf(layer), 1);
+        setLayers([...layers]);
+    }
+
+    function removeAllTracks() {
+        layers.forEach(l => map.removeLayer(l.layer));
+        setLayers([]);
+    }
+
+    function addTrackToMap(track) {
+        let layer = new L.GPX(track.gpx, {
+            async: true,
+            marker_options: markerOptions
+        }).on('loaded', function (e) {
+            map.fitBounds(e.target.getBounds());
+        }).addTo(map);
+        setLayers([...layers, {name: track.name, layer: layer}])
+    }
+
+    useEffect(() => {
+        if (ctx.localClientsTracks.length === 0) {
```

Comment:
this code is not correct, it won't work once it will be feature to delete 1 track (currenlty it works because you can delete only all tracks)

Reply:
`+`